src/main/java/org/jenkinsci/plugins/workflow/cps/LoggingInvoker.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.cps;
+
+import com.cloudbees.groovy.cps.impl.CallSiteBlock;
+import com.cloudbees.groovy.cps.sandbox.DefaultInvoker;
+import com.cloudbees.groovy.cps.sandbox.Invoker;
+import com.cloudbees.groovy.cps.sandbox.SandboxInvoker;
+import groovy.lang.GroovyClassLoader;
+import java.util.function.Supplier;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Captures CPS-transformed events.
+ */
+final class LoggingInvoker implements Invoker {
+
+    public static Invoker create(boolean sandbox) {
+        return new LoggingInvoker(sandbox ? new SandboxInvoker() : new DefaultInvoker());
+    }
+
+    private final Invoker delegate;
+
+    private LoggingInvoker(Invoker delegate) {
+        this.delegate = delegate;
+    }
+
+    private void record(String call) {
+        CpsThreadGroup g = CpsThreadGroup.current();
+        if (g == null) {
+            assert false : "should never happen";
+            return;
+        }
+        CpsFlowExecution execution = g.getExecution();
+        if (execution == null) {
+            assert false : "should never happen";
+            return;
+        }
+        execution.recordInternalCall(call);
+    }
+
+    private static boolean isInternal(Class<?> clazz) {
+        if (clazz == Safepoint.class || clazz == CpsClosure2.class) {
+            return false;
+        }
```

Comment:
I think `CpsDefaultGroovyMethods` should be ignored here as well.

I ran this plugin's tests with a modification to fail every build where `getInternalCalls` is non-empty to look for false positives. Other than security-related tests that call `Jenkins.get`, the only failure was `CpsFlowDefinition2Test.sandboxInvokerUsed`, which also tracked `com.cloudbees.groovy.cps.CpsDefaultGroovyMethods.$callClosureForMapEntry__groovy_lang_Closure__java_util_Map_Entry`  and `com.cloudbees.groovy.cps.CpsDefaultGroovyMethods.$collectEntries__java_util_Map__java_util_Map__groovy_lang_Closure`.